zebra-network/src/peer_set/set.rs

```diff
@@ -423,6 +423,12 @@ where
                     .ready_services
                     .get_index_mut(index)
                     .expect("preselected index must be valid");
+
+                // `PeerSet` calls `poll_ready` multiple times on its `Client` services.
+                // `Buffer`ed polls can fill up their Buffer reservations, causing hangs.
+                // The next `assert` will make sure the service type is never `Buffer`. #1593
```

Comment:
We can move part of the explanation into an assert message:
```suggestion
                // See #1593 for details.
```